common/theme_generator.py

```diff
@@ -70,10 +70,13 @@ def get_new_theme_path(self, name):
 
         full_path = os.path.join(sublime.packages_path(), path_in_packages)
 
-        with open(full_path, "wb") as out_f:
-            out_f.write(STYLES_HEADER.encode("utf-8"))
-            out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))
-
+        try:
+            with open(full_path, "wb") as out_f:
```

Comment:
Searching the code, there are a 16 other calls to `with open()`, perhaps we should write a wrapping context manager function to handle open and catch `OSError` exceptions for all of them?

Most of the other file actions are on repo files, so it differs from #634 but could have a similar impact of "failing strangely".